### ISSUE TITLE
Simplify/Remove the compilation process

### ISSUE THREAD

**Issue by mtreca** [emacsorphanage/gnuplot#52]
So, looking through the files of this repository, I see 8 different compilation or installation-related files (excluding `INSTALL.org`). They also are pretty consequent files:

```sh
$ wc -l aclocal.m4 configure configure.ac install-sh Makefile.am Makefile.dst Makefile.in missing 
   700 aclocal.m4
  3730 configure
    19 configure.ac
   527 install-sh
    40 Makefile.am
    39 Makefile.dst
   729 Makefile.in
   215 missing
  5999 total
```

So the question is, do we really need all these files? I have never maintained an emacs package before but I am pretty sure most of these files are useless to some degree (even when manually installing this package).

Looking at the Makefiles in more details, I see that installing this package is a three step process:

- Byte compiling the elisp files
- Building the TeX reference card
- Moving info files to the info path

I don't think a makefile is needed for any of these steps:

- AFAIK, byte-compiling should be done automatically when installing gnuplot through any package manager. Users installing the package manually should then byte compile manually (this will be included in the installation instructions).
- I think the reference card should be provided in pdf format directly.
- Moving the provided info files has to be done manually in the current setup anyways, so the Makefile adds no benefits here.

@tarsius I would like to now what you think about this situation. Would it be safe to simply remove all these Makfiles? What would the manual installation instructions look like without a Makefile?

**Comment by bruceravel** [comment]
All the build stuff was written loooooong before Emacs had a proper package manager. :smile:

**Comment by vxid** [comment]
@bruceravel Looking at the NEWS section of the README, this seems about right!
So do you think most of these files should be dropped? I was thinking that keeping `Makefile.dst` should be sufficient...


Thanks for this package by the way, I use it daily for my research work!

**Comment by tarsius** [comment]
Having a basic Makefile is useful even now, but this is of course extreme. So yeah, its probably best to throw out what we have now and start over.

Over the years I have come up with a way to write Makefiles for all my packages. It's basically all boilerplate and the individual Makefiles differ only slightly. Actually there's two basic variants.

* Look at [moody/Makefile](https://github.com/tarsius/moody/blob/master/moody.el) for the simple variant that just compiles and generates the autoload file.
* Based on that see [epkg/Makefile](https://github.com/emacscollective/epkg/blob/master/Makefile) for the variant that also generates and published a manual.

I recommend you copy and adjust moody's variant. After that you can copy the refcard related targets from the current ball of build files. My extened variant is not useful here.

(And I will cleanup epkg's variant. That's the problem with using a template, one has to keep the results manually after coming up with a better way of doing something and sometimes one forgets. For example the "silencio" hacks that are still used here are not useful anymore.)

**Comment by mtreca** [comment]
Closed to closed with 90bfd6d236fe4ce55c219216d8c36970552d3ed8, I will still need to port most of @tarsius' Makefile templates to the new Makefile.

**Comment by mtreca** [comment]
Fixed in 1dd661e
